test/test_skill_intents.py

```diff
@@ -0,0 +1,266 @@
+
+from ovos_core.intent_services import PadatiousService, PadatiousMatcher
+from ovos_bus_client import Message
+from ovos_bus_client.session import Session, SessionManager
+from ovos_config.config import update_mycroft_config, Configuration
+from ovos_utils.messagebus import FakeBus
+from ovos_utils.log import LOG
+from ovos_plugin_manager.skills import find_skill_plugins
+from ovos_workshop.skill_launcher import SkillLoader
+from ovos_workshop.skills.base import BaseSkill
+
+
+PIPELINE = ["adapt_high", "adapt_medium", "adapt_low"] 
+use_padacioso = getenv("INTENT_ENGINE") == "padacioso"
+if use_padacioso:
+    PIPELINE.extend(["padacioso_high",
+                     "padacioso_medium",
+                     "padacioso_low"])
+else:
+    PIPELINE.extend(["padatious_high",
+                     "padatious_medium",
+                     "padatious_low"])
+LOG.level = logging.DEBUG
+
+
+class MockPadatiousMatcher(PadatiousMatcher):
+    include_med = True
+    include_low = False
+
+    def __init__(self, *args, **kwargs):
+        PadatiousMatcher.__init__(self, *args, **kwargs)
+        LOG.debug("Creating test Padatious Matcher")
+
+    def match_medium(self, utterances, lang=None, __=None):
+        if not self.include_med:
+            LOG.info(f"Skipping medium confidence check for {utterances}")
+            return None
+        PadatiousMatcher.match_medium(self, utterances, lang=lang)
+
+    def match_low(self, utterances, lang=None, __=None):
+        if not self.include_low:
+            LOG.info(f"Skipping low confidence check for {utterances}")
+            return None
+        PadatiousMatcher.match_low(self, utterances, lang=lang)
+
+
+def get_skill_object(skill_entrypoint: str, bus: FakeBus,
+                     skill_id: str, config_patch: Optional[dict] = None) -> BaseSkill:
+    """
+    Get an initialized skill object by entrypoint with the requested skill_id.
+    @param skill_entrypoint: Skill plugin entrypoint or directory path
+    @param bus: FakeBus instance to bind to skill for testing
+    @param skill_id: skill_id to initialize skill with
+    @param config_patch: Configuration update to apply
+    @returns: Initialized skill object
+    """
+    if config_patch:
+        user_config = update_mycroft_config(config_patch)
+        if user_config not in Configuration.xdg_configs:
+            Configuration.xdg_configs.append(user_config)
+    if isdir(skill_entrypoint):
+        LOG.info(f"Loading local skill: {skill_entrypoint}")
+        loader = SkillLoader(bus, skill_entrypoint, skill_id)
+        if loader.load():
+            return loader.instance
+    plugins = find_skill_plugins()
+    if skill_entrypoint not in plugins:
+        raise ValueError(f"Requested skill not found: {skill_entrypoint}; available skills: {list(plugins.keys())}")
+    plugin = plugins[skill_entrypoint]
+    skill = plugin(bus=bus, skill_id=skill_id)
+    return skill
+
+
+class TestSkillIntentMatching(unittest.TestCase):
```

Comment:
This appears to be a functionally equivalent to [neon-minerva](https://github.com/NeonGeckoCom/neon-minerva/blob/dev/neon_minerva/tests/test_skill_intents.py). consider install/import to reduce duplicated code